log10/load.py

```diff
@@ -267,6 +268,68 @@ def log_sync(completion_url, func, **kwargs):
     return completionID
 
 
+class StreamingResponseWrapper:
+    def __init__(self, completion_url, completionID, response, partial_log_row):
+        self.completionID = completionID
+        self.completion_url = completion_url
+        self.partial_log_row = partial_log_row
+        self.response = response
+        self.final_result = ""  # Store the final result
+        self.start_time = time.perf_counter()
+        self.gpt_id = None
+        self.model = None
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            chunk = next(self.response)
+            content = chunk.choices[0].delta.content
+            if content:
+                # Here you can intercept and modify content if needed
+                self.final_result += content  # Save the content
+                # Yield the original or modified content
+
+                self.model = chunk.model
+                self.gpt_id = chunk.id
+
+            return chunk
+        except StopIteration as se:
+            # Log the final result
+            # Create fake response for openai format.
+            response = {
+                "id": self.gpt_id,
+                "object": "completion",
+                "model": self.model,
+                "choices": [
+                    {
+                        "index": 0,
+                        "finish_reason": "stop",
+                        "message": {
+                            "role": "assistant",
+                            "content": self.final_result,
+                        },
+                    }
+                ],
+            }
+            self.partial_log_row["response"] = json.dumps(response)
+            self.partial_log_row["duration"] = int(time.perf_counter() - self.start_time * 1000)
+
+            try:
+                res = post_request(self.completion_url + "/" + self.completionID, self.partial_log_row)
+                if res.status_code != 200:
+                    logger.error(f"LOG10: failed to insert in log10: {self.partial_log_row} with error {res.text}")
+            except Exception as e:
+                traceback.print_tb(e.__traceback__)
+                logging.warn(f"LOG10: failed to log: {e}. Skipping")
+
+            raise se
+
+    def get_final_result(self):
+        return self.final_result
```

Comment:
curious is this been called somewhere now or just a util function? Do we need to check StopIterator or has been raised or some state telling that this is the final results?

Reply:
We can remove it